store/store_download.go

```diff
@@ -162,36 +161,14 @@ func (s *Store) cdnHeader() (string, error) {
 		return "none", nil
 	}
 
-	if s.dauthCtx == nil {
-		return "", nil
-	}
-
 	// set Snap-CDN from cloud instance information
 	// if available
 
 	// TODO: do we want a more complex retry strategy
 	// where we first to send this header and if the
 	// operation fails that way to even get the connection
 	// then we retry without sending this?
-
-	cloudInfo, err := s.dauthCtx.CloudInfo()
-	if err != nil {
-		return "", err
-	}
-
-	if cloudInfo != nil {
-		cdnParams := []string{fmt.Sprintf("cloud-name=%q", cloudInfo.Name)}
-		if cloudInfo.Region != "" {
-			cdnParams = append(cdnParams, fmt.Sprintf("region=%q", cloudInfo.Region))
-		}
-		if cloudInfo.AvailabilityZone != "" {
-			cdnParams = append(cdnParams, fmt.Sprintf("availability-zone=%q", cloudInfo.AvailabilityZone))
-		}
-
-		return strings.Join(cdnParams, " "), nil
-	}
-
-	return "", nil
+	return s.buildLocationString()
```

Comment:
now download requests will have both Snap-CDN and Snap-Device-Location, I don't know if we care presently to phase the older one out?

Reply:
As I mentioned in the commit message, I think we will probably end up leaving `Snap-CDN` for only the `Snap-CDN: none` case. Once the new snapd is out in the wild I'll make the store use `Snap-Device-Location` if `Snap-CDN` isn't present.

Reply:
sorry, it seems a skipped reading the summary closely when I reviewed this